test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -68,8 +68,7 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
 	validations := f.Helper().ValidationSetForUnsupportedFeatureSet(unsupportedFeatures)
 
 	BeforeEach(func() {
-		acmeIssuer := util.NewCertManagerACMEIssuer(issuerName, f.Config.Addons.ACMEServer.URL, testingACMEEmail, testingACMEPrivateKey)
```

Comment:
am I correct these changes are related to:

> I have removed a bunch of legacy functions for issuer generation ...

if so, I can see why that's a valuable change but this is already a pretty huge PR and these changes to use `gen` here feel quite distant from the goal of this specific PR; they should maybe be in their own PR, to make everything easier to review?

or have I missed how these changes are linked to this PR?

Reply:
I put it here because I was adding some e2e tests for issuers and without removing the legacy I would have to either use & update the legacy functions for issuer generation or have two different ways of doing this in one file.
I will create a separate PR with these changes.

Reply:
#3873 PRs in just the removal of legacy functions.